runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -529,35 +556,33 @@ void J9::RecognizedCallTransformer::processUnsafeAtomicCall(TR::TreeTop* treetop
 
          }
 
+      // ramStatics is not collected. We cannot reuse objectNode->getSymbolReference()
+      newSymbolReference = comp()->getSymRefTab()->createTemporary(comp()->getMethodSymbol(), TR::Address);
+      newSymbolReference->getSymbol()->setNotCollected();
+
       // Calculate static address
       auto objectAdjustmentNode = TR::Node::createWithSymRef(TR::astore, 1, 1,
                                                              TR::Node::createWithSymRef(TR::aloadi, 1, 1,
                                                                                         TR::Node::createWithSymRef(TR::aloadi, 1, 1,
                                                                                                                    objectNode->duplicateTree(),
                                                                                                                    comp()->getSymRefTab()->findOrCreateClassFromJavaLangClassSymbolRef()),
                                                                                         comp()->getSymRefTab()->findOrCreateRamStaticsFromClassSymbolRef()),
-                                                             objectNode->getSymbolReference());
+                                                             newSymbolReference);
       auto offsetAdjustmentNode = TR::Node::createWithSymRef(TR::lstore, 1, 1,
                                                              TR::Node::create(TR::land, 2,
                                                                               offsetNode->duplicateTree(),
                                                                               TR::Node::lconst(~J9_SUN_FIELD_OFFSET_MASK)),
                                                              offsetNode->getSymbolReference());
 
       if (enableTrace)
-         traceMsg(comp(), "Created node n%dn and n%dn to adjust object and offset for static field\n", objectAdjustmentNode->getGlobalIndex(), offsetAdjustmentNode->getGlobalIndex());
+         traceMsg(comp(), "Created node objectAdjustmentNode n%dn #%d and offsetAdjustmentNode #%d n%dn to adjust object and offset for static field\n",
+               objectAdjustmentNode->getGlobalIndex(), objectAdjustmentNode->getSymbolReference() ? objectAdjustmentNode->getSymbolReference()->getReferenceNumber() : -1,
```

Comment:
The preferred way to check if a node has a sym ref or not is via `hasSymbolReference` rather than `getSymbolReference`. It used to be possible that there happens to be a sym ref still present in the node even though the opcode was changed to one that may not have a sym ref.

Reply:
Addressed in 072048a